lib/src/utils.dart

```diff
@@ -2,7 +2,12 @@
 // for details. All rights reserved. Use of this source code is governed by a
 // BSD-style license that can be found in the LICENSE file.
 
+import 'package:charcode/charcode.dart';
+import 'package:tuple/tuple.dart';
```

Comment:
We want to avoid new dependencies on this package. How about writing a `SubSpan` class?

Reply:
A "subspan" isn't really identified as a pair of locations... this is just a multiple return. Is there really no general-purpose way of doing that anymore?

Reply:
We don't currently have a general purpose way of retuning multiple values in our core packages or any packages in the SDK. We are using small single use classes for now, and will hopefully eventually replace them with language supported tuples.